odoo/addons/test_orm/tests/test_schema.py

```diff
@@ -26,152 +27,13 @@ def assertSelectionXID(self, record):
-            'test_orm.model_selection_related_updatable',
-            'test_orm.model_selection_required',
-            'test_orm.model_selection_required_for_write_override',
-            'test_orm.model_shared_cache_compute_line',
-            'test_orm.model_shared_cache_compute_parent',
-            'test_orm.modified',
-            'test_orm.modified.line',
-            'test_orm.monetary_base',
-            'test_orm.monetary_custom',
-            'test_orm.monetary_inherits',
-            'test_orm.monetary_order',
-            'test_orm.monetary_order_line',
-            'test_orm.monetary_related',
-            'test_orm.move',
-            'test_orm.move_line',
-            'test_orm.multi',
-            'test_orm.multi.line',
-            'test_orm.multi.line2',
-            'test_orm.multi.tag',
-            'test_orm.multi_compute_inverse',
-            'test_orm.onchange.partial.view',
-            'test_orm.one2many',
-            'test_orm.one2many.line',
-            'test_orm.order',
-            'test_orm.order.line',
-            'test_orm.partner',
-            'test_orm.payment',
-            'test_orm.person',
-            'test_orm.person.account',
-            'test_orm.pirate',
-            'test_orm.precompute',
-            'test_orm.precompute.combo',
-            'test_orm.precompute.editable',
-            'test_orm.precompute.line',
-            'test_orm.precompute.monetary',
-            'test_orm.precompute.readonly',
-            'test_orm.precompute.required',
-            'test_orm.prefetch',
-            'test_orm.prefetch.line',
-            'test_orm.prisoner',
-            'test_orm.recursive',
-            'test_orm.recursive.line',
-            'test_orm.recursive.order',
-            'test_orm.recursive.task',
-            'test_orm.recursive.tree',
-            'test_orm.related',
-            'test_orm.related_bar',
-            'test_orm.related_foo',
-            'test_orm.related_inherits',
-            'test_orm.related_translation_1',
-            'test_orm.related_translation_2',
-            'test_orm.related_translation_3',
-            'test_orm.req_m2o',
-            'test_orm.req_m2o_transient',
-            'test_orm.selection',
-            'test_orm.shared.compute',
-            'test_orm.ship',
-            'test_orm.state_mixin',
-            'test_orm.team',
-            'test_orm.team.member',
-            'test_orm.transient_model',
-            'test_orm.trigger.left',
-            'test_orm.trigger.middle',
-            'test_orm.trigger.right',
-            'test_orm.unsearchable.o2m',
-            'test_orm.user',
-            'test_orm.view.str.id',
-        ]
-        ir_models = self.env['ir.model'].search([('model', 'in', model_names)])
+        model_names = {'domain.bool', 'decimal.precision.test'}
+
+        for cls in MetaModel._module_to_models__["test_orm"]:
+            if cls._name.startswith('test_orm'):
+                model_names.add(cls._name)
```

Comment:
I don't see the point of using a `set` here, except for introducing a little bit of nondeterminism 😄  Actually because the set of domains is only used in a domain, you don't have to worry about duplicates.  Make it a plain stupid list.